src/Features/CSharp/Portable/Completion/CompletionProviders/LastBuiltInCompletionProvider.cs

```diff
@@ -0,0 +1,25 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+using System.Composition;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Completion;
+
+namespace Microsoft.CodeAnalysis.CSharp.Completion.Providers
+{
+    /// <summary>
+    /// Provides a completion provider that always appears after all built-in completion providers. This completion
+    /// provider does not provide any completions.
```

Comment:
what is this for then? is this just for testing purposes?

Reply:
Completion providers are a public API. This allows users to provide a completion provider at the beginning or end of the order.